packages/typir/src/kinds/class-kind.ts

```diff
@@ -562,7 +571,7 @@ export class ClassTypeInitializer<T = unknown, T1 = unknown, T2 = unknown> exten
         this.initialClassType = new ClassType(kind, typeDetails as CreateClassTypeDetails);
         if (kind.options.typing === 'Structural') {
             // register structural classes also by their names, since these names are usually used for reference in the DSL/AST!
-            this.services.graph.addNode(this.initialClassType, kind.getIdentifierPrefix() + typeDetails.className);
+            this.services.graph.addNode(this.initialClassType, kind.calculateNominalIdentifier(typeDetails));
```

Comment:
It is structural, but uses the nominal identifier?

Reply:
I renamed the function, since its name was misleading: When used in languages (e.g. TypeScript), structurally typed classes are used by other classes not by encoding their structure but by using their "names". Therefore "name-based identifiers" are relevant for structurally typed classes as well.